### PR TITLE
Further refactoring to avoid @user

### PR DESCRIPTION
**Pull request by gravitystorm**
This follows on from #1583 and refactors the remaining controller code, and all of the views, to use the `current_user` approach for accessing the currently logged in user.

The `self.current_user =` feels a bit clumsy, but it's necessary to avoid creating local variables with the same name as the controller method. I'm sure those will disappear when we eventually migrate to a devise or similar.

### CODE REVIEW

File: routes.txt [hunk 1]
```diff
@@ -0,0 +1,270 @@
+                               Prefix Verb     URI Pattern                                                  Controller#Action
```
Comment (tomhughes) [inline routes.txt]: Did you intend to add this file?

Comment (gravitystorm) [inline routes.txt]: D'oh.

Comment (gravitystorm) [inline routes.txt]: OK, force-updated the PR without that file now.

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -47,7 +49,7 @@ def require_user
   end
 
   def require_oauth
-    @oauth = @user.access_token(OAUTH_KEY) if @user && defined? OAUTH_KEY
+    @oauth = @user.access_token(OAUTH_KEY) if current_user && defined? OAUTH_KEY
```
Comment (tomhughes) [inline app/controllers/application_controller.rb]: You missed one in this line ;-)

As far as I could tell last night that was the only one in any view or controller that had been missed, which is probably pretty good doing.

Comment (gravitystorm) [inline app/controllers/application_controller.rb]: Oh dear, I really thought I'd found them all!

**Pull request merged**